src/OrchardCore.Modules/OrchardCore.Contents/Recipes/ContentStep.cs

```diff
@@ -49,6 +49,8 @@ public async Task ExecuteAsync(RecipeExecutionContext context)
                     _session.Save(existing);
                 }
             }
+
+            await _session.CommitAsync();
```

Comment:
I think that you don't need this line anymore if you merge the dev branch where we now commit the session automatically before disposing the shell scope.

Reply:
I can remove it but I don't think it will cause any issue calling it early right?

Reply:
BTW I don't think this PR has anything to do with the ContentStep recipes.

Reply:
> BTW I don't think this PR has anything to do with the ContentStep recipes.

Yes not directly but the issue was to do CommitAsync in session Dispose that may use an already disposed scoped service, so it may happen in different contexts and was happening during a setup recipe.

> I can remove it but I don't think it will cause any issue calling it early right?

Yes but maybe worth to re-try some setups without this line and i think that now the right pattern is to let the session being committed automatically (and once) just before the shell scope is disposed, this after any session save that may occur here or after.

Reply:
Let's open an issue and a PR to keep track of that specific issue.